wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/jpa/JpaFactoryDao.java

```diff
@@ -92,7 +96,7 @@ public void remove(String id) throws ServerException {
   }
 
   @Override
-  @Transactional
+  @Transactional(rollbackOn = {RuntimeException.class, ServerException.class})
```

Comment:
I think `@Transactional(rollbackOn = ServerException.class)` would be enough. But it is better to have more than needed but not less than needed in this case )

Reply:
In this case, will transaction be rolled on runtime exc ?

Reply:
Ah you mean that runtime is catched here.ok